src/core/src/engine.rs

```diff
@@ -345,23 +341,22 @@ impl Engine for Poet2Engine {
 
             if !published_at_height && time::Instant::now().duration_since(start) > wait_time {
                 info!("Timer expired -- publishing block");
+                info!("wait time was : {:?} for chain head: {:?}", wait_time, chain_head.clone());
```

Comment:
We might not want to log the complete block here. BlockId should rather suffice. Also update the log level to debug maybe.

Reply:
We need to see the whole block for now to confirm the state of the block.